src/ebfm/INIT.py

```diff
@@ -328,6 +328,79 @@ def init_grid(grid, io, config: GridConfig):
+
+        # loop over the azimuth angles to determine gridded maximum grid angles per angle
+        grid["maxgridangle"] = np.zeros((grid["gpsum"], grid["nr_az_steps"]), dtype=np.float64)
+        for n in range(grid["nr_az_steps"]):
+            az = np.full(int(grid["gpsum"]), grid["az_array"][n], dtype=float)
+
+            # calculate step sizes (ddx, ddy) in x- and y-directions for all azimuth angles
+            ddx = np.empty_like(az, dtype=float)
+            ddy = np.empty_like(az, dtype=float)
+            m = az <= -0.75 * np.pi
+            ddx[m] = -np.tan(np.pi + az[m])
+            m = (az <= -0.25 * np.pi) & (az > -0.75 * np.pi)
+            ddx[m] = -1.0
+            m = (az <= 0.25 * np.pi) & (az > -0.25 * np.pi)
+            ddx[m] = np.tan(az[m])
+            m = (az <= 0.75 * np.pi) & (az > 0.25 * np.pi)
+            ddx[m] = 1.0
+            m = az > 0.75 * np.pi
+            ddx[m] = np.tan(np.pi - az[m])
+            m = az <= -0.75 * np.pi
+            ddy[m] = 1.0
+            m = (az <= -0.25 * np.pi) & (az > -0.75 * np.pi)
+            ddy[m] = -np.tan(0.5 * np.pi + az[m])
+            m = (az <= 0.25 * np.pi) & (az > -0.25 * np.pi)
+            ddy[m] = -1.0
+            m = (az <= 0.75 * np.pi) & (az > 0.25 * np.pi)
+            ddy[m] = -np.tan(0.5 * np.pi - az[m])
+            m = az > 0.75 * np.pi
+            ddy[m] = 1.0
+
+            # from every grid cell step in the direction of the azimuth until the grid end is reached
+            # and detect maximum grid angle along the path
+            i0, j0 = np.where(mask_2D == 1)
+            max_angle = np.full(grid["gpsum"], -np.inf, dtype=np.float64)
+            count = 1
+            active = np.ones(grid["gpsum"], dtype=bool)
+            while active.any():
+                j = np.round(j0 + ddx * count).astype(np.int64)
+                i = np.round(i0 + ddy * count).astype(np.int64)
+
+                inbound = (j >= 0) & (j < yl) & (i >= 0) & (i < xl) & active
+                if not inbound.any():  # stop when all walks have reached the domain edge
+                    break
+
+                iv = i[inbound]
+                jv = j[inbound]
+
+                dx = grid["x_2D"][iv, jv] - grid["x"][inbound]
+                dy = grid["y_2D"][iv, jv] - grid["y"][inbound]
+                distance = np.hypot(dx, dy)  # walk distance from start to target
+
+                dz = grid["z_2D"][iv, jv] - grid["z"][inbound]
+                grid_angle = np.arctan(dz / distance)  # grid angle from start to target
+
+                max_angle[inbound] = np.maximum(max_angle[inbound], grid_angle)  # update max grid angle when needed
+
+                active &= (j >= 0) & (j < yl) & (i >= 0) & (i < xl)  # continue walk until domain edge is reached
+
+                count += 1
+
+            # fill lookup table with maximum grid angles for all cells (dimension 1) and azimuth angle (dimension 2)
+            grid["maxgridangle"][:, n] = max_angle
```

Comment:
If we use the squared distance above we would have to apply a correction here:

```suggestion
            grid["maxgridangle"][:, n] = np.arctan(np.sqrt(np.tan(max_angle_proxy)))
```

This costs more than a `sqrt`, but we are saving a lot of `sqrt`s above. So it should pay off. But: let's not do it if it is not measurable.